opensearchpy/connection/http_requests.py

```diff
@@ -187,9 +189,17 @@ def perform_request(  # type: ignore
             "allow_redirects": allow_redirects,
         }
         send_kwargs.update(settings)
+
+        calculate_service_time = False
+        if "calculate_service_time" in self.kwargs:
+            calculate_service_time = self.kwargs["calculate_service_time"]
+
+        time_metrics = TimeMetrics()
+
         try:
+            time_metrics.events.server_request_start()
```

Comment:
You shouldn't b reaching inside `.events`, it should be something like `self.metrics.request_start()`.